hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -818,9 +818,9 @@ public boolean merge(MapMergeTypes mergingEntry,
 
             record = createRecord(key, newValue, UNSET, UNSET, now);
             mergeRecordExpiration(record, mergingEntry);
-            newValue = persistenceEnabledFor(provenance)
-                    ? mapDataStore.add(key, newValue, record.getExpirationTime(), now, null) : newValue;
-            recordFactory.setValue(record, newValue);
+            if (persistenceEnabledFor(provenance)) {
+                runMapStore(record, key, newValue, now, null);
```

Comment:
Don't we need to update the `newValue` here since it might affect the return value of the method?
`newValue = runMapStore(record, key, newValue, now, null);`

Reply:
seems fine since there is no usage which requires updated `newValue`